manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -138,10 +138,10 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 				if t.Status.State >= api.TaskStateOrphaned && t.ServiceID == "" {
 					tr.cleanup = append(tr.cleanup, t.ID)
 				}
-				// add tasks that have progressed beyond SHUTDOWN and have desired state REMOVE. These
+				// add tasks that have progressed beyond COMPLETE and have desired state REMOVE. These
 				// tasks are associated with slots that were removed as part of a service scale down
 				// or service removal.
-				if t.DesiredState == api.TaskStateRemove && t.Status.State >= api.TaskStateShutdown {
+				if t.DesiredState == api.TaskStateRemove && t.Status.State >= api.TaskStateCompleted {
```

Comment:
What about the tasks which reached completed state before we removed/downsized the service ?

Reply:
They'll be handled by the regular task retention logic, won't they?

Reply:
I realized that the reaper will still see task updates for those tasks when the desired state is updated to REMOVE.